riptide-compression/src/main/java/org/zalando/riptide/compression/Compression.java

```diff
@@ -0,0 +1,23 @@
+package org.zalando.riptide.compression;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apiguardian.api.API;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+@API(status = EXPERIMENTAL)
+@AllArgsConstructor(staticName = "of")
+@Getter
+public final class Compression {
+    private final String contentEncoding;
+    private final OutputStreamWrapper outputStreamWrapper;
```

Comment:
`ThrowingUnaryOperator<OutputStream, IOException>`? :thinking:

Reply:
I am personally afraid of `@SneakyThrows` but I am happy to use `ThrowingUnaryOperator` here if this aligns with other usages in riptide :-)

Reply:
Adapted.